src/main/java/org/kiwiproject/dropwizard/util/health/MonitoredJobHealthCheck.java

```diff
@@ -0,0 +1,143 @@
+            var lastFailure = job.getLastFailure().get();
+            if ((now - lastFailure) < errorWarningDuration.toMilliseconds()) {
+                return buildUnhealthyResult(f("An error has occurred at: {}, which is within the threshold of: {}",
+                        instantToStringOrNever(job.getLastFailure().get()), errorWarningDuration));
+            }
+
+            var warningThreshold = getWarningThreshold();
+            if ((now - getTimeOrServerStart(lastRun)) > warningThreshold) {
+                return buildUnhealthyResult(f("Last successful execution was: {}, which is older than the threshold of: {} seconds",
+                        instantToStringOrNever(lastRun), Duration.milliseconds(warningThreshold).toSeconds()));
+            }
+
+            return buildHealthyResult(f("Last successful execution was: {}", instantToStringOrNever(lastRun)));
+        } catch (Exception e) {
+            LOG.error("Encountered Exception: ", e);
+            return handleException(e);
+        }
+    }
+
+    private Result buildHealthyResult(String message) {
+        return resultBuilderWith(message, true).build();
+    }
+
+    private ResultBuilder resultBuilderWith(String message, boolean healthy) {
+        return resultBuilderWith(message, healthy, null);
+    }
+
+    private ResultBuilder resultBuilderWith(String message, boolean healthy, Exception error) {
+        checkValidHealthArgumentCombination(healthy, error);
+        var resultBuilder = isNull(error) ? newResultBuilder(healthy)
+                : newUnhealthyResultBuilder(error);
+
+        return resultBuilder
+                .withMessage(message)
+                .withDetail("jobName", job.getName())
+                .withDetail("totalErrors", job.getFailureCount())
+                .withDetail("lastFailure", job.getLastFailure())
+                .withDetail("lastSuccess", job.getLastSuccess())
+                .withDetail("lastExecutionTimeMs", job.getLastExecutionTime())
+                .withDetail("expectedFrequencyMs", expectedFrequency.toMilliseconds())
+                .withDetail("warningThresholdMs", getWarningThreshold())
+                .withDetail("errorWarningDurationMs", errorWarningDuration.toMilliseconds());
+    }
+
+    private static void checkValidHealthArgumentCombination(boolean healthy, Exception error) {
+        checkArgument(!healthy || isNull(error), "If healthy, error must be null!");
+    }
+
+    private long getWarningThreshold() {
+        return Math.max((long) (expectedFrequency.toMilliseconds() * thresholdFactor),
+                MINIMUM_WARNING_THRESHOLD.toMilliseconds());
+    }
+
+    private static String instantToStringOrNever(long epochMs) {
+        return epochMs != 0 ? Instant.ofEpochMilli(epochMs).toString() : "never";
+    }
+
+    private Result buildUnhealthyResult(String message) {
+        return resultBuilderWith(message, false).build();
+    }
+
+    private long getTimeOrServerStart(long lastRun) {
```

Comment:
Suggest adding `Ms` as a suffix like you did in several other places, e.g. in the `instantToStringOrNever` method above where the argument name is `epochMs`

Reply:
If you are talking about line 79, the `toString` of Duration adds the units

Reply:
Oh nm, you mean in the variable name.